tools/replace_code_block.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import sys
+from traceback import print_tb
+from typing import AnyStr
+
+status = 0
+
+
+def check_for_code(file: AnyStr, check: bool):
+    """Check a file and replace ``.. code::`` to ``.. code-block::``
+
+    :param file: The file to open
+    :param check: Whether or not the program is operating in ``check`` mode.
+        In check mode, the program will exit with a status 1 if files will be
+        replaced. In non-check mode, the program will replace files.
+    """
+    global status
+    with open(file, "rb") as infile:
+        content = infile.read()
+    old_content = content
+    new_content = content.replace(b".. code::", b".. code-block::")
+    if old_content != new_content:
+        if check:
+            status = 1
+            print(
+                "File {filename} contains ``code::`` statements!".format(
+                    filename=file
+                )
+            )
+        else:
+            with open(file, "wb") as outfile:
+                outfile.write(new_content)
+            print(
+                "Replaced ``code::`` statements to ``code-block::`` in "
+                "{filename}".format(filename=file)
+            )
+
+
+def main(check):
+    global status
+    for current_directory, folder_list, file_list in os.walk(
+        os.path.abspath(os.path.join(__file__, "..", "..", "praw"))
+    ):
+        for file in file_list:
+            file = "{dir}/{filename}".format(
+                dir=current_directory, filename=file
+            )
+            if file[-3:] == ".py":
+                try:
+                    check_for_code(file, check)
+                except Exception:
```

Comment:
Did you actually encounter any exceptions? If not, I'd say let's not explicitly handle them. If so, let's catch those specific ones.

Reply:
There should be no exceptions. And yes, I did encounter an exception.

Reply:
So if there _should_ be no exceptions, why are we handling it?

Reply:
If there is an exception, I don't want the entire program to break, and it throws a different statuscode

Reply:
Since this tool is intended for PRAW developers, if there is an exception we want it to break things. Also generally, unless your running code that is intended never to quit it's not a great idea to catch the general `Exception` class.

Reply:
I guess we can omit the bit then.